From: Miklos Szeredi <mszeredi@suse.cz>
Date: Wed, 17 Apr 2013 21:50:58 +0200
Subject: fuse: optimize wake_up
Git-commit: 3c18ef8117f0515b1e455713dfc2e18b06db9bb5
Patch-mainline: v3.10-rc1
References: FATE#317677

Normally blocked_waitq will be inactive, so optimize this case.

Signed-off-by: Miklos Szeredi <mszeredi@suse.cz>
---
 fs/fuse/dev.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/fs/fuse/dev.c
+++ b/fs/fuse/dev.c
@@ -380,7 +380,7 @@ __releases(fc->lock)
 			fc->blocked = 0;
 
 		/* Wake up next waiter, if any */
-		if (!fc->blocked)
+		if (!fc->blocked && waitqueue_active(&fc->blocked_waitq))
 			wake_up(&fc->blocked_waitq);
 
 		if (fc->num_background == fc->congestion_threshold &&
